sys/posix/sockets/posix_sockets.c

```diff
@@ -279,16 +284,42 @@ static int socket_close(int socket)
     return res;
 }
 
-static ssize_t socket_read(int socket, void *buf, size_t n)
+static inline int socket_fstat(vfs_file_t *filp, struct stat *buf)
+{
+    (void)filp;
+    memset(buf, 0, sizeof(struct stat));
+    buf->st_mode |= (S_IFSOCK | S_IRWXU | S_IRWXG | S_IRWXO);
+    buf->st_blksize = 512;
+    return 0;
+}
+
+static inline off_t socket_lseek(vfs_file_t *filp, off_t off, int whence)
 {
-    return recv(socket, buf, n, 0);
+    (void)filp;
+    (void)off;
+    (void)whence;
+    return -ESPIPE; /* see http://pubs.opengroup.org/onlinepubs/9699919799/functions/lseek.html */
 }
 
-static ssize_t socket_write(int socket, const void *buf, size_t n)
+static inline ssize_t socket_read(vfs_file_t *filp, void *buf, size_t n)
 {
-    return send(socket, buf, n, 0);
+    return socket_recvfrom(filp->private_data.ptr, buf, n, 0, NULL, NULL);
 }
 
+static inline ssize_t socket_write(vfs_file_t *filp, const void *buf, size_t n)
+{
+    return socket_sendto(filp->private_data.ptr, buf, n, 0, NULL, 0);
+}
+
+static const vfs_file_ops_t socket_ops = {
+    .close = socket_close,
+    .fcntl = NULL,          /* TODO: provide when needed */
```

Comment:
maybe provide an inline function that does some debug output? Just incase its used in the future?